object/datastore.go

```diff
@@ -99,13 +99,17 @@ func (d Datastore) Browser(ctx context.Context) (*HostDatastoreBrowser, error) {
 // ServiceTicket obtains a ticket via AcquireGenericServiceTicket and returns it an http.Cookie with the url.URL
 // that can be used along with the ticket cookie to access the given path.
 func (d Datastore) ServiceTicket(ctx context.Context, path string, method string) (*url.URL, *http.Cookie, error) {
+	dcPath := "ha-datacenter"
+	if d.InventoryPath != "" {
+		dcPath = strings.Split(d.InventoryPath, "/")[1]
```

Comment:
This will only work if you are connected directly to ESX, otherwise the VC datacenter can have any name.  Maybe for now we could use and environment variable to override, "GOVMOMI_ESX_DATACENTER" that defaults to "ha-datacenter".  In any case, we should find out why "ha-datacenter" was renamed to "datacenter" in the dev version, and if it's going to stay way we can find a better solution.